client/lib/abtest/index.js

```diff
@@ -349,6 +350,9 @@ ABTest.prototype.saveVariation = function ( variation ) {
 		this.recordVariation( variation );
 	}
 	this.saveVariationInLocalStorage( variation );
+
+	const experimentId = this.experimentId.replace( /[A-Z]/g, ( s ) => `_${ s.toLowerCase() }` );
```

Comment:
@taggon  Is there advantage to be had if this was moved to `recordVariation()` for client-side tracking, and then duplicated in the server-side API call; i.e., following logic in the comment above?

> // Note that for logged-in users, we fire the Tracks event from the API abtest endpoint
> // to ensure that the numbers match up exactly with the user attributes data
> // to ensure that the numbers match up exactly with the user attributes data

Reply:
Good point. Although it isn't clear to me how the described issue happened in the past, we might be well staying consistent with it. @taggon What do you think?

Reply:
@southp @jaswrks I removed the altered experiment ID for consistency. Originally I thought it is necessary for consistency because all the other `bumpStat` event ID are snake case. :)

Reply:
@taggon Good call to keep the experiment ID consistent. However, there is one more consistency concern here that I'd be grateful if you can help us find out, about the comment @jaswrks found in this line https://github.com/Automattic/wp-calypso/blob/master/client/lib/abtest/index.js#L346.

It's concerning because it implies that we should also bump the stats from the backend for logged-in users. Could you help us verify if it's a concern or not?

Reply:
@jaswrks @southp I don't think we should care about the comment. The purpose of the block is to get the exact number of users who are involved in A/B tests by checking the user attributes data. Also, it allows us to find the users from the server-side even if they are assigned to an a/b test in Calypso.

Reply:
@taggon Thanks! That leads me to different question. Can this `bumpStat()` call potentially fire repeatedly for the same user; e.g., when they come back through the flow again? If so, should we try to avoid this? I'm trying to imagine how we can best use the bump stats, and it seems like one important difference between Tracks and Bump Stats is that Tracks is recording the user's anon/user ID, so we can see unique users. Whereas bump stats aren't?

Reply:
>  Can this `bumpStat()` call potentially fire repeatedly for the same user; e.g., when they come back through the flow again?

@jaswrks I don't think so. The function `saveVariation()`, that calls `bumpStat()`, is only invoked when the user is not assigned to any variation of the target a/b test. Please check the following line.
https://github.com/Automattic/wp-calypso/blob/master/client/lib/abtest/index.js#L172

Reply:
I see. Thanks, Taegon! 👍